### PR TITLE
Fix IOU Options bug

### PR DESCRIPTION
**Pull request by sobitneupane**
Show Request and Send Money options when there are two participants, Splitbill option when there are more than two participants and no option if there is only one participant

### Fixed Issues

$ https://github.com/Expensify/App/issues/8357

### Tests

1. Open one-to-one chats, group chats and chat rooms with single participant, two participants and multiple participants
2. Click on Action (+) Button in ReportAction Compose box to see various options.
3. Verify that 
     * there are "Request money", "Send money" and "Add attachment" options in one-to-one chats and chat rooms with two participants.
     * there are "Split bill" and "Add attachment" options in group chats and chat rooms with more than two participants. 
     * there is only "Add attachment" option in chat room with single participant.

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Open one-to-one chats, group chats and chat rooms with single participant, two participants and multiple participants
2. Click on Action (+) Button in ReportAction Compose box to see various options.
3. Verify that 
     * there are "Request money", "Send money" and "Add attachment" options in one-to-one chats and chat rooms with two participants.
     * there are "Split bill" and "Add attachment" options in group chats and chat rooms with more than two participants. 
     * there is only "Add attachment" option in chat room with single participant.

- [x] Verify that no errors appear in the JS console

#### Web

| Chat Room With only 1 member | Chat Room With 2 members | Chat Room With more than 2 members |
:-------------------------:|:-------------------------:|:-------------------------:
|  |  |  |

| One-to-One Chat | Group Chat |
:-------------------------:|:-------------------------:
|  |  | 

#### Mobile Web

| Chat Room With only 1 member | Chat Room With 2 members | Chat Room With more than 2 members |
:-------------------------:|:-------------------------:|:-------------------------:
|Screenshot_1651500869 |Screenshot_1651500838|Screenshot_1651500800|

| One-to-One Chat | Group Chat |
:-------------------------:|:-------------------------:
| Screenshot_1651500815| Screenshot_1651500825| 

#### Desktop

| Chat Room With only 1 member | Chat Room With 2 members | Chat Room With more than 2 members |
:-------------------------:|:-------------------------:|:-------------------------:
|  |  |  |

| One-to-One Chat | Group Chat |
:-------------------------:|:-------------------------:
|  |  | 

#### iOS

| Chat Room With only 1 member | Chat Room With 2 members | Chat Room With more than 2 members |
:-------------------------:|:-------------------------:|:-------------------------:
|Simulator Screen Shot - iPhone 13 - 2022-05-02 at 20 05 21|Simulator Screen Shot - iPhone 13 - 2022-05-02 at 20 05 06|Simulator Screen Shot - iPhone 13 - 2022-05-02 at 20 04 56|

| One-to-One Chat | Group Chat |
:-------------------------:|:-------------------------:
|Simulator Screen Shot - iPhone 13 - 2022-05-02 at 20 04 36| Simulator Screen Shot - iPhone 13 - 2022-05-02 at 20 04 45| 

#### Android

| Chat Room With only 1 member | Chat Room With 2 members | Chat Room With more than 2 members |
:-------------------------:|:-------------------------:|:-------------------------:
|Screenshot_1651500423] |Screenshot_1651500186| Screenshot_1651500296|

| One-to-One Chat | Group Chat |
:-------------------------:|:-------------------------:
|Screenshot_1651500118|Screenshot_1651500144|

### CODE REVIEW

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Can you check out this https://github.com/Expensify/App/issues/8357#issuecomment-1109974388

I've asked to move the logic to a new method!

File: src/pages/home/report/ReportActionCompose.js
Comment (Santhosh-Sellavel) [inline src/pages/home/report/ReportActionCompose.js:245]: Do we really need to compute and pass everything to this function it does not make any sense?

File: src/pages/home/report/ReportActionCompose.js
Comment (Santhosh-Sellavel) [inline src/pages/home/report/ReportActionCompose.js:300]: Code is not DRY, we have redundant checks here.  Can you refactor this?

**Comment by sobitneupane** [comment]
> Can you check out this [#8357 (comment)](https://github.com/Expensify/App/issues/8357#issuecomment-1109974388)
> 
> I've asked to move the logic to a new method!

Done.

@iwiznia @Santhosh-Sellavel I have found another existing bug. While splitting bill, the logged in user(Chaudary Dai in this case) is counted twice.
Screen Shot 2022-05-02 at 23 01 11
 
 I think changing 
 ```
 participants={this.props.hasMultipleParticipants
                                                     ? this.state.participants
                                                     : _.filter(this.state.participants, email => this.props.myPersonalDetails.login !== email.login)}
```
to 
 ```
 participants={_.filter(this.state.participants, email => this.props.myPersonalDetails.login !== email.login)}
```
in IOUModal.js solves this issue too.

Should I go through whole process of reporting? Or will I be compensated if I solved it here?

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/pages/iou/IOUModal.js
Comment (Santhosh-Sellavel) [inline src/pages/iou/IOUModal.js:434]: Should this be done here, I think it could be moved along the lines here https://github.com/Expensify/App/blob/4c64bec3f38af925a01911155eab54feafc96cda/src/pages/iou/IOUModal.js#L110-L111

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Comment by Santhosh-Sellavel** [comment]
@sobitneupane I believe this is well within the scope of this issue.
If it's going to get compensated then it's for reporting only. It was evidently reported by someone else before you in this https://github.com/Expensify/App/issues/8357#issuecomment-1113707406 

cc @iwiznia

**Comment by sobitneupane** [comment]
> @sobitneupane I believe this is well within the scope of this issue. If it's going to get compensated then it's for reporting only. It was evidently reported by someone else before you in this [#8357 (comment)](https://github.com/Expensify/App/issues/8357#issuecomment-1113707406)

I have already solved the issue mentioned in this comment while solving this issue as it is part of this issue. But while testing I found that the issue I mentioned also occurs. If it is believed to be part of this issue I will solve it here.

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js:300]: Refactoring this might result into nested ternary operators. I was discouraged to use nested ternary operators in previous PRs. Any suggestion?

**Comment by Santhosh-Sellavel** [comment]
> > @sobitneupane I believe this is well within the scope of this issue. If it's going to get compensated then it's for reporting only. It was evidently reported by someone else before you in this [#8357 (comment)](https://github.com/Expensify/App/issues/8357#issuecomment-1113707406)
> 
> I have already solved the issue mentioned in this comment while solving this issue as it is part of this issue. It would have been an issue introduced while solving this issue. But while testing I found that the issue I mentioned also occurs. If it is believed to be part of this issue I will solve it here. It was an existing issue.

Thanks, I would let @iwiznia take the final call!

File: src/pages/home/report/ReportActionCompose.js
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionCompose.js:300]: We could use simple if/else no need for ternary here.

File: src/pages/iou/IOUModal.js
Comment (sobitneupane) [reply src/pages/iou/IOUModal.js:434]: Participants are updated during IOUParicipantsPage step. So, it cannot be moved here. We need to make use of updated participants list.

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js:301]: It will be something like this. Any suggestion?
```
        const actions = [];
        if (!hasExcludedIOUEmails) {
            if (Permissions.canUseIOU(this.props.betas)) {
                if (showSplitBill) {
                    actions.push(
                        {
                            icon: Expensicons.Receipt,
                            text: this.props.translate('iou.splitBill'),
                            onSelected: () => {
                                Navigation.navigate(
                                    ROUTES.getIouSplitRoute(
                                        this.props.reportID,
                                    ),
                                );
                            },
                        },
                    );
                }
                if (showSendRequestMoney) {
                    actions.push({
                        icon: Expensicons.MoneyCircle,
                        text: this.props.translate('iou.requestMoney'),
                        onSelected: () => {
                            Navigation.navigate(
                                ROUTES.getIouRequestRoute(
                                    this.props.reportID,
                                ),
                            );
                        },
                    });
                }
            }
            if (Permissions.canUseIOUSend(this.props.betas) && showSendRequestMoney) {
                actions.push({
                    icon: Expensicons.Send,
                    text: this.props.translate('iou.sendMoney'),
                    onSelected: () => {
                        Navigation.navigate(
                            ROUTES.getIOUSendRoute(
                                this.props.reportID,
                            ),
                        );
                    },
                });
            }
        }

        actions.push({
            icon: Expensicons.Paperclip,
            text: this.props.translate('reportActionCompose.addAttachment'),
            onSelected: () => {
                openPicker({
                    onPicked: (file) => {
                        displayFileInModal({file});
                    },
                });
            },
        });
        return actions;
```

File: src/pages/iou/IOUModal.js
Comment (Santhosh-Sellavel) [reply src/pages/iou/IOUModal.js:434]: `IOUParicipantsPage` will not be used for Room or group flow. So no issues there.

Comment (Santhosh-Sellavel) [reply src/pages/iou/IOUModal.js:434]: Check and let me know!

File: src/pages/home/report/ReportActionCompose.js
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionCompose.js:301]: Let's just move IOU options computation alone to a new method, leave Add Attachment untouched thanks!

```js
 getIOUOptions(reportParticipants) {
        const iouOptions = [];
        const participants = _.filter(reportParticipants, email => this.props.myPersonalDetails.login !== email).length > 1;
        const hasExcludedIOUEmails = lodashIntersection(reportParticipants, CONST.EXPENSIFY_EMAILS).length > 0;
        const hasMultipleParticipants = participants.length > 1;

        if (hasExcludedIOUEmails || participants.length === 0 || Permissions.canUseIOU(this.props.betas)) {
            return iouOptions;
        }

        if (hasMultipleParticipants) {
            // append split
        }
        else {
            if (Permissions.canUseIOUSend(this.props.betas)) {
                // append send
            }
            // append request
        }

        return iouOptions;
    }
```

cc: @iwiznia

File: src/pages/iou/IOUModal.js
Comment (sobitneupane) [reply src/pages/iou/IOUModal.js:434]: I don't think it's a good idea. It adds more complication.

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: Done.

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: Resolved.

**Comment by sobitneupane** [comment]
> > > @sobitneupane I believe this is well within the scope of this issue. If it's going to get compensated then it's for reporting only. It was evidently reported by someone else before you in this [#8357 (comment)](https://github.com/Expensify/App/issues/8357#issuecomment-1113707406)
> > 
> > 
> > I have already solved the issue mentioned in this comment while solving this issue as it is part of this issue. It would have been an issue introduced while solving this issue. But while testing I found that the issue I mentioned also occurs. If it is believed to be part of this issue I will solve it here. It was an existing issue.
> 
> Thanks, I would let @iwiznia take the final call!

Solved this issue as well. I am okay either way.

**Comment by iwiznia** [comment]
I think it's basically the same issue or very closely related, so original amount should apply. If you disagree, please let me know and we can discuss.

File: src/pages/home/report/ReportActionCompose.js
Comment (iwiznia) [inline src/pages/home/report/ReportActionCompose.js:247]:
```suggestion
            return [];
```

File: src/pages/home/report/ReportActionCompose.js
Comment (iwiznia) [inline src/pages/home/report/ReportActionCompose.js:265]:
```suggestion
            const iouOptions = [];
            iouOptions.push({
```

**Review by iwiznia — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: made the requested change

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: made the requested change

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: made the requested change

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js:264]: else statement is removed as it was not needed and variable is declared at the top of the function.

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: made the requested change

File: src/pages/home/report/ReportActionCompose.js
Comment (sobitneupane) [reply src/pages/home/report/ReportActionCompose.js]: made the requested change

**Review by iwiznia — DISMISSED** [review]

**Comment by iwiznia** [comment]
@Santhosh-Sellavel if everything looks good, please approve the PR.

File: src/pages/home/report/ReportActionCompose.js
Comment (Santhosh-Sellavel) [inline src/pages/home/report/ReportActionCompose.js:639]: prop type is defined for `myPersonalDetails` can you add that, thanks!

**Comment by Santhosh-Sellavel** [comment]
#### PR Reviewer Checklist

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

**Comment by Santhosh-Sellavel** [comment]
@sobitneupane 
Add what to verify as result step with options to show based on different chat types. Add it to the Tests & QA Test sections.

Example:
In 1:1 verify that only request, send, attachment options are shown ..

**Review by Santhosh-Sellavel — APPROVED** [review]
@iwiznia 
I have left a comment [here](https://github.com/Expensify/App/pull/8839#issuecomment-1117704850) to update the test steps.

Everything else looks good, tests well!

**Review by iwiznia — APPROVED** [review]

**Comment by iwiznia** [comment]
Waiting till the comment above is addressed before merging

**Comment by sobitneupane** [comment]
> @sobitneupane Add what to verify as result step with options to show based on different chat types. Add it to the Tests & QA Test sections.

Added.

**Comment by Santhosh-Sellavel** [comment]
@iwiznia We are good to go here!

**Pull request merged**
